service/parserprovider/configsource.go

```diff
@@ -0,0 +1,93 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package parserprovider
+
+import (
+	"context"
+	"fmt"
+
+	"go.uber.org/zap"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/experimental/configsource"
+	"go.opentelemetry.io/collector/experimental/configsource/envvarconfigsource"
+	"go.opentelemetry.io/collector/experimental/configsource/fileconfigsource"
+	"go.opentelemetry.io/collector/experimental/configsource/vaultconfigsource"
+)
+
+type configSourceParserProvider struct {
```

Comment:
Creates a parser provider that completes hides the manager from service and application.